### PR TITLE
fix: simplify RedisHashTags

### PR DESCRIPTION
**Pull request by jasonschroeder-sfdc**
Fixes: #1686

### CODE REVIEW

File: src/main/java/build/buildfarm/common/redis/RedisHashtags.java [hunk 1]
```diff
@@ -31,8 +35,10 @@ public class RedisHashtags {
    * @return A valid queue name for one of the internal queues.
    * @note Suggested return identifier: queueName.
    */
-  public static String hashedName(String name, String hashtag) {
-    return "{" + hashtag + "}" + name;
+  public static String hashedName(@Nonnull String name, @Nullable String hashtag) {
```
Comment (werkt) [inline src/main/java/build/buildfarm/common/redis/RedisHashtags.java:38]: we assume that non-@Nullable fields are not accepted to be null - definitely have some outliers, but I don't want to start going down the route of tagging *everything* one way or the other.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 2]
```diff
@@ -1497,16 +1497,16 @@ public boolean canPrequeue() throws IOException {
...
-    builder.setPrequeue(state.prequeue.status(client.call(jedis -> jedis)));
-    return builder.build();
+    return BackplaneStatus.newBuilder()
+        .addAllActiveExecuteWorkers(executeWorkers)
+        .addAllActiveStorageWorkers(storageWorkers)
+        .addAllActiveWorkers(Sets.union(executeWorkers, storageWorkers))
+        .setDispatchedSize(client.call(jedis -> state.dispatchedOperations.size(jedis)))
+        .setOperationQueue(state.operationQueue.status(client.call(jedis -> jedis)))
+        .setPrequeue(state.prequeue.status(client.call(jedis -> jedis)))
+        .build();
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:1509]: Aside from the formatting change, I don't see any material modifications to this. Does BackplaneStatus now do something different?

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 3]
```diff
@@ -1497,16 +1497,16 @@ public boolean canPrequeue() throws IOException {
...
-    builder.setDispatchedSize(client.call(jedis -> state.dispatchedOperations.size(jedis)));
-    builder.setOperationQueue(state.operationQueue.status(client.call(jedis -> jedis)));
-    builder.setPrequeue(state.prequeue.status(client.call(jedis -> jedis)));
-    return builder.build();
+    return BackplaneStatus.newBuilder()
+        .addAllActiveExecuteWorkers(executeWorkers)
+        .addAllActiveStorageWorkers(storageWorkers)
+        .addAllActiveWorkers(Sets.union(executeWorkers, storageWorkers))
+        .setDispatchedSize(client.call(jedis -> state.dispatchedOperations.size(jedis)))
+        .setOperationQueue(state.operationQueue.status(client.call(jedis -> jedis)))
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:1507]: Apropos of nothing, but... what? This is yielding the jedis instance, that is returned to the pool - this is asking for a desync failure...
I know it's not related to your change, but I cannot believe this was in there.
This needs to be `client.call(jedis -> state.operationQueue.status(jedis))` and likewise for the following line with prequeue.

**Review by werkt — CHANGES_REQUESTED** [review]
If this changes BackplaneStatus to return names without hashes, then I need to see that differentiated from the metrics presentation, as there's no other way to present the redis data structures used for queues per their hash positioning within the key.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 3 above)
Comment (jasonschroeder-sfdc) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:1507]: OK if I make that change in a separate PR?

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 2 above)
Comment (jasonschroeder-sfdc) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:1509]: No, just style change.

**Comment by jasonschroeder-sfdc** [comment]
> If this changes BackplaneStatus to return names without hashes, then I need to see that differentiated from the metrics presentation, as there's no other way to present the redis data structures used for queues per their hash positioning within the key.

This only seems to affect when the hash is empty string.

The name is mangled here:
https://github.com/bazelbuild/bazel-buildfarm/blob/e1ad3ad22875a2462d19db36727b498d9e738f81/src/main/java/build/buildfarm/common/redis/BalancedRedisQueue.java#L377-L382

```java
379         setName(RedisHashTags.hashedName("cpu", ""))
```

returns:
```
{}cpu
```
after the change, it returns:
```
cpu
```


**If this isn't the right place for a fix, I can do a `RedisHashTags.unhashedName(String)` near the metrics presentation. I can create a separate PR if you'd like to see that option 😄**

File: src/main/java/build/buildfarm/common/redis/RedisHashtags.java (hunk 1 above)
Comment (jasonschroeder-sfdc) [reply src/main/java/build/buildfarm/common/redis/RedisHashtags.java:38]: No worries, I can revert all of the `javax.annotations.*` in this class.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 3 above)
Comment (jasonschroeder-sfdc) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:1507]: Moved to https://github.com/bazelbuild/bazel-buildfarm/pull/1691

**Comment by werkt** [comment]
> **If this isn't the right place for a fix, I can do a `RedisHashTags.unhashedName(String)` near the metrics presentation. I can create a separate PR if you'd like to see that option 😄**

That's what I'd prefer - mangle this downstream if this is the structure consumed by metrics. There has to be some way to get the queue name out - I'd actually prefer if we could associate each of the queue responses in the status with a name in redis, and then also offer the hashed-template name for the pool (or maybe skip it if we had each constituent).

**Comment by jasonschroeder-sfdc** [comment]
Here's a smaller attempt. https://github.com/bazelbuild/bazel-buildfarm/pull/1693

**Pull request closed without merging**